### PR TITLE
Routing fixes

### LINKED ISSUE #22: After reserve one file successfully, resubmit a new file
I was not able to come back to the reserve submission steps from the attached screen.

Screen Shot 2021-02-09 at 1 28 34 PM

### LINKED ISSUE #52: Implement URL handling to reflect "pages"
Update web pages to include URLs to enable some basic state for the webpages

### Acceptance Critieria
* Able to use a static URL to return to a page in the workflow (does not need to be super robust, does not have to be mid-action)
* Use navigation buttons forward / backward (does not need to be super robust)

### LINKED ISSUE #64: As an administrator, I want to deploy the application in an apache server under a subdirectory/subpath
## Motivation
...so that I can deploy the application for example in /var/www/dois and access it from http://pds-gamma.jpl.nasa.gov/dois 

## Acceptance Criteria
**Given** an apache server
**When I perform** cp -r build <apache documentroot>/dois
**Then I expect** the application to work from url http://{localhost}/dois

### LINKED ISSUE #65: As a user, I want my search criteria to persist in the URL
## Motivation
...so that I can easily share or come back to a specific DOI search.

## Acceptance Criteria
**Given** a deployed DOI service with data ingested
**When I perform** a search for a DOI like `*insight*`
**Then I expect** the URL to include the API query parameters

## Engineering Details

Follow on to #52

### PR DESCRIPTION
**Pull request by eddiesarevalo**
Fixed the issues caused by merging #22 #52 #57 together. The pull requests had to be updated with the router code.
This also adds #65 through #52

resolves #22 
resolves #52 
resolves #57 
resolves #65 
resolves #64 

I need @c-suh to verify that the latest changes are there and functional. I tested end to end with a dummy excel doi. @c-suh can use her tests to double check everything is working.

### CODE REVIEW

**Review by tloubrieu-jpl — CHANGES_REQUESTED** [review]
That is a big one, thanks Eddie for putting that together.

Only change I would ask for is the button "Reserve a new DOI" should be "create a new DOI" or "create another DOI" (since we are asking the question again `Has the data been registered and made publicly available?` in the next step, the user can reserve or draft/release a DOI).

Otherwise that sounds good to me. I will wait for @c-suh 's review before I merge.

Thanks,

File: src/components/SearchResults.jsx
Comment (c-suh) [inline src/components/SearchResults.jsx:279]: Since the majority of this is commented out, this whole chunk could be `null` or its contents deleted.

**Review by c-suh — COMMENTED** [review]
This is looking good, @eddiesarevalo ! I did notice a few things and pushed a change for an overlooked placeholder link.

1. Search behavior: IIRC, the latest decision regarding the default view for the [Search page ](http://localhost:3000/search) is that it's not supposed to display any results until the search button (or enter from the search bar) is entered.
2. Search behavior: clicking the 'X' in the search bar to clear the search - should this clear the results as well as the text in the search bar?
3. There's a chunk of code in SearchResults.jsx (lines 187-279) which is mostly comments. Either the commented out part can be deleted or those lines can be replaced with `null`.

**Comment by c-suh** [comment]
@tloubrieu-jpl, I left a review [above](https://github.com/NASA-PDS/pds-doi-ui/pull/66#pullrequestreview-703463063), but you might want to sign off on my points 1 and 2 as "fine as is"?

**Comment by tloubrieu-jpl** [comment]
Thanks @c-suh , I will create new tickets for your comments so that I can merge now. Thanks

**Comment by tloubrieu-jpl** [comment]
Hi @eddiesarevalo,

Were you able to change the buton label to create a new DOI after one is reserved ?

Thanks,

Thomas

**Comment by eddiesarevalo** [comment]
@tloubrieu-jpl I changed the label for the button. I also changed the basename subdiretory to '/dois'

**Comment by tloubrieu-jpl** [comment]
thanks @eddiesarevalo , is the basename subdiretory hardcoded ? 

for the production deployment we want to deploy the build application in an apache directory and ideally the application would be agnostic of which apache directory it is deployed in.

**Comment by eddiesarevalo** [comment]
@tloubrieu-jpl Try to test this latest commit. It should allow deployment on any sub directory. I have only been able to test locally but you can try it on your server. 

Use `npm run build` to get the built code.

**Comment by tloubrieu-jpl** [comment]
Thanks @eddiesarevalo that works. There is only a minor issue that does not prevent from merging and deploying on pds-gamma. I created a ticket #70

**Pull request merged**
